forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java

```diff
@@ -134,17 +134,27 @@ public void resolve(SpellAbility sa) {
             name = hostCard + (sa.hasParam("Boon") ? "'s Boon" : "'s Effect");
         }
 
-        // Unique Effects shouldn't be duplicated
-        if (sa.hasParam("Unique") && game.isCardInCommand(name)) {
-            return;
-        }
-
         if (sa.hasParam("EffectOwner")) {
             effectOwner = AbilityUtils.getDefinedPlayers(hostCard, sa.getParam("EffectOwner"), sa);
         } else {
             effectOwner = Lists.newArrayList(sa.getActivatingPlayer());
         }
 
+        // Unique$ is for effects that should be one per player (e.g. Gollum, Obsessed Stalker)
+        if (sa.hasParam("Unique")) {
+            List<Player> remove = null;
```

Comment:
this isn't initialized and you call add on it
your IDE should warn...

if you turn it into a PlayerCollection you could also call `for (Player eo : effectOwner.threadSafeIterable())` on it